javacutil/src/main/java/org/checkerframework/javacutil/TypesUtils.java

```diff
@@ -248,9 +248,25 @@ public static boolean isIntegral(TypeMirror type) {
         }
     }
 
+    /**
+     * Returns true iff the argument is a boxed floating point type.
+     *
+     * @param type type mirror
+     * @return whether the argument is a boxed floating point type
+     */
+    public static boolean isBoxedFloating(TypeMirror type) {
+        if (type.getKind() != TypeKind.DECLARED) {
+            return false;
+        }
+
+        String qualifiedName = getQualifiedName((DeclaredType) type).toString();
```

Comment:
At some point we should figure out and document what the proper way is. `Name` defines a `contentEquals`, which might be more direct than calling `toString`. However, for multiple comparisons one approach might be more efficient.

Reply:
That does sound like a good future improvement.